src/main/java/org/opensearch/searchrelevance/settings/SearchRelevanceSettings.java

```diff
@@ -41,4 +41,16 @@ public class SearchRelevanceSettings {
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
+
+    /**
+     * Gates the maximum limit of search relevance query set size
+     * By defaulted, we set the value as 150
+     */
+    public static final String SEARCH_RELEVANCE_QUERY_SET_MAX_LIMIT_KEY = "plugins.search_relevance.query_set.maximum";
+    public static final Setting<Integer> SEARCH_RELEVANCE_QUERY_SET_MAX_LIMIT = Setting.intSetting(
+        SEARCH_RELEVANCE_QUERY_SET_MAX_LIMIT_KEY,
+        150,
```

Comment:
150 sounds like a bit small number, is there a rationale behind this or it has been picked at random?